### PR TITLE
Deprecate `useDeleteButtonTooltip` for Chips

### PR DESCRIPTION
**Pull request by royarg02**
This PR deprecates [`useDeleteButtonTooltip`](https://api.flutter.dev/flutter/material/DeletableChipAttributes/useDeleteButtonTooltip.html), for `Chip` widgets that support having a delete button. This includes the `Chip`, the `InputChip` and the ` RawChip` widget.

Instead, to disable the tooltips on the delete button of these chips, users should pass an **empty string** to [`deleteButtonTooltipMessage`](https://api.flutter.dev/flutter/material/DeletableChipAttributes/deleteButtonTooltipMessage.html).

A [dart fix](https://dart.dev/tools/dart-fix) is provided.

The migration code is shown below:

**Before:**
```dart
Chip(useDeleteButtonTooltip: false);
InputChip(useDeleteButtonTooltip: true);
RawChip rawChip = RawChip();
rawChip.useDeleteButtonTooltip;
```

**After:**
```dart
Chip(deleteButtonTooltipMessage: ‘’);
InputChip();
RawChip rawChip = RawChip();
rawChip.deleteButtonTooltipMessage;
```

The design doc for this change can be found [here](https://flutter.dev/go/chip-usedeletebuttontooltip-deprecation).

This is not a breaking change as it is only deprecating a property.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/fix_data.yaml
Comment (RoyARG02) [inline packages/flutter/lib/fix_data.yaml:17]:
```suggestion
  # Changes made in https://github.com/flutter/flutter/pull/96174
```

File: packages/flutter/lib/fix_data.yaml
Comment (RoyARG02) [inline packages/flutter/lib/fix_data.yaml:42]:
```suggestion
  # Changes made in https://github.com/flutter/flutter/pull/96174
```

File: packages/flutter/lib/fix_data.yaml
Comment (RoyARG02) [inline packages/flutter/lib/fix_data.yaml:67]:
```suggestion
  # Changes made in https://github.com/flutter/flutter/pull/96174
```

File: packages/flutter/lib/fix_data.yaml
Comment (RoyARG02) [inline packages/flutter/lib/fix_data.yaml:92]:
```suggestion
  # Changes made in https://github.com/flutter/flutter/pull/96174
```

File: packages/flutter/lib/fix_data.yaml
Comment (RoyARG02) [inline packages/flutter/lib/fix_data.yaml:103]:
```suggestion
  # Changes made in https://github.com/flutter/flutter/pull/96174
```

File: packages/flutter/lib/fix_data.yaml
Comment (RoyARG02) [inline packages/flutter/lib/fix_data.yaml:114]:
```suggestion
  # Changes made in https://github.com/flutter/flutter/pull/96174
```

File: packages/flutter/lib/src/material/chip.dart
Comment (RoyARG02) [inline packages/flutter/lib/src/material/chip.dart:254]: TODO: Add flutter version

File: packages/flutter/lib/src/material/chip.dart
Comment (RoyARG02) [inline packages/flutter/lib/src/material/chip.dart:578]: TODO: Add flutter version

File: packages/flutter/lib/src/material/chip.dart
Comment (RoyARG02) [inline packages/flutter/lib/src/material/chip.dart:628]: TODO: Add flutter version

File: packages/flutter/lib/src/material/chip.dart
Comment (RoyARG02) [inline packages/flutter/lib/src/material/chip.dart:758]: TODO: Add flutter version

File: packages/flutter/lib/src/material/chip.dart
Comment (RoyARG02) [inline packages/flutter/lib/src/material/chip.dart:835]: TODO: Add flutter version

File: packages/flutter/lib/src/material/chip.dart
Comment (RoyARG02) [inline packages/flutter/lib/src/material/chip.dart:1515]: TODO: Add flutter version

File: packages/flutter/lib/src/material/chip.dart
Comment (RoyARG02) [inline packages/flutter/lib/src/material/chip.dart:1593]: TODO: Add flutter version

File: packages/flutter/test_fixes/material.dart
Comment (RoyARG02) [inline packages/flutter/test_fixes/material.dart:525]:
```suggestion
  // Changes made in https://github.com/flutter/flutter/pull/96174
```

File: packages/flutter/test_fixes/material.dart
Comment (RoyARG02) [inline packages/flutter/test_fixes/material.dart:532]:
```suggestion
  // Changes made in https://github.com/flutter/flutter/pull/96174
```

File: packages/flutter/test_fixes/material.dart
Comment (RoyARG02) [inline packages/flutter/test_fixes/material.dart:539]:
```suggestion
  // Changes made in https://github.com/flutter/flutter/pull/96174
```

File: packages/flutter/test_fixes/material.dart.expect
Comment (RoyARG02) [inline packages/flutter/test_fixes/material.dart.expect:498]:
```suggestion
  // Changes made in https://github.com/flutter/flutter/pull/96174
```

File: packages/flutter/test_fixes/material.dart.expect
Comment (RoyARG02) [inline packages/flutter/test_fixes/material.dart.expect:505]:
```suggestion
  // Changes made in https://github.com/flutter/flutter/pull/96174
```

File: packages/flutter/test_fixes/material.dart.expect
Comment (RoyARG02) [inline packages/flutter/test_fixes/material.dart.expect:512]:
```suggestion
  // Changes made in https://github.com/flutter/flutter/pull/96174
```

**Comment by RoyARG02** [comment]
@gspencergoog The PR is (mostly) ready for review, I just have a few concerns:
1. Since dev channel is deprecated, what version should I provide for the deprecation notices?
2. I updated some comments and a test for delete button tooltip, which I committed here https://github.com/RoyARG02/flutter/commit/bf86261edd101d67a6bd09f2cb9c5a335da117c2. If it looks good to you, I'll submit a separate PR.

**Review by Piinks — COMMENTED** [review]
Woohoo! Thanks for writing a dart fix! 💯 
For the deprecation version, update with the latest from the master branch, and then use the version from `flutter doctor`.

**Comment by RoyARG02** [comment]
Hey @Piinks, I applied the fixes on a test project, and I'm getting the following behavior:

```dart
RawChip(
    label: const Text('No tooltip for this delete button!'),
    onDeleted: (){},
    useDeleteButtonTooltip: false,
)
```
Becomes
```dart
RawChip(
    label: const Text('No tooltip for this delete button!'),
    onDeleted: (){}, deleteButtonTooltipMessage: '',
)
```

Is there something I need to add so that the parameter is added to its own line?

**Review by Piinks — COMMENTED** [review]
> Is there something I need to add so that the parameter is added to its own line?

I don't think this is possible, the dart fix tool does not affect formatting in this way. @bwilkerson can you confirm?

**Review by Piinks — APPROVED** [review]
Nice! Thank you for the updates, this LGTM!

**Comment by Piinks** [comment]
This will need a second review, I've reached out on discord to find someone. :)

**Review by Hixie — APPROVED** [review]
LGTM

**Comment by Hixie** [comment]
Looks like there are some conflicts to resolve before it can land, unfortunately.

**Pull request merged**
